app/core/models.py

```diff
@@ -1,3 +1,137 @@
+from django.utils.translation import gettext_lazy as _
 from django.db import models
+import django.contrib.auth.models as auth_models
+from django.utils import timezone
+from django.urls import reverse
 
-# Create your models here.
+from . import configurator
+
+
+class Timestamped(models.Model):
+    created_at = models.DateTimeField(default=timezone.now)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        abstract = True
+        get_latest_by = ("created_at",)
+
+
+class Challenge(Timestamped):
+    name = models.CharField(max_length=255)
+    start_at = models.DateTimeField()
+    end_at = models.DateTimeField()
+    repo_url = models.URLField()
+    # Data stored in S3 - privileges on S3 buckets will help
+    # prevent leakage of secret data
+    sample_data_url = models.URLField()
+    sample_score_reference_url = models.URLField()
+    secret_data_url = models.URLField()
+    secret_score_reference_url = models.URLField()
+    execution_options_json = models.JSONField()
+    # TODO: How to store scoring container reference in challenge?
```

Comment:
Can we associate an instance of our Container model?

Reply:
One option:
- scoring_container - 'nullable'
- create the challenge
- create the container, associate to the challenge
- set the scoring container in the challenge

OR
- Container could be a top level, independent of Challenge
- Challenge has a nullable scoring_container fkey to container
- Relationship between submission, challenge, and container lives only in submission

```
             Scoring Container
                                 \
Container               Challenge
              \                /
               submission
```

Reply:
I added a 'scoremaker' model to address this